src/Microsoft.AspNetCore.SignalR.Client/HubConnection.cs

```diff
@@ -134,15 +132,26 @@ public void On(string methodName, Type[] parameterTypes, Action<object[]> handle
                 _logger.LogTrace("Invocation #{0}: {1} {2}({3})", descriptor.Id, returnType.FullName, methodName, argsList);
             }
 
-            var ms = new MemoryStream();
-            await _adapter.WriteMessageAsync(descriptor, ms, cancellationToken);
+            try
+            {
+                var ms = new MemoryStream();
+                await _adapter.WriteMessageAsync(descriptor, ms, cancellationToken);
 
-            _logger.LogInformation("Sending Invocation #{0}", descriptor.Id);
+                _logger.LogInformation("Sending Invocation #{0}", descriptor.Id);
 
-            // TODO: Format.Text - who, where and when decides about the format of outgoing messages
-            // TODO HIGH: Handle return value/Exception from SendAsync
-            await _connection.SendAsync(ms.ToArray(), MessageType.Text, cancellationToken);
-            _logger.LogInformation("Sending Invocation #{0} complete", descriptor.Id);
+                // TODO: Format.Text - who, where and when decides about the format of outgoing messages
+                await _connection.SendAsync(ms.ToArray(), MessageType.Text, cancellationToken);
+                _logger.LogInformation("Sending Invocation #{0} complete", descriptor.Id);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(0, ex, "Sending Invocation #{0} failed", descriptor.Id);
+                irq.Completion.SetException(ex);
```

Comment:
this needs to be TrySetException - the result can be set in the meantime if the connection closes.